stdlib/experimental/http/requests/requests.go

```diff
@@ -0,0 +1,214 @@
+		}
+		config := configV.Object()
+
+		var body io.Reader
+		if bodyV, ok := args.Get("body"); ok {
+			if bodyV.Type().Nature() != semantic.Bytes {
+				return nil, errors.Newf(codes.Invalid, "parameter \"body\" is not of type bytes: %v", bodyV.Type())
+			}
+			body = bytes.NewReader(bodyV.Bytes())
+		}
+
+		// Construct HTTP request
+		req, err := http.NewRequestWithContext(ctx, method, u.String(), body)
+		if err != nil {
+			return nil, err
+		}
+
+		// Add headers to request
+		headersV, ok := args.Get("headers")
+		if ok && !headersV.IsNull() {
+			if headersV.Type().Nature() != semantic.Dictionary {
+				return nil, errors.Newf(codes.Invalid, "parameter \"headers\" is not of type [string:string] : %v", headersV.Type())
+			}
+			var rangeErr error
+			headersV.Dict().Range(func(k values.Value, v values.Value) {
+				if k.Type().Nature() == semantic.String &&
+					v.Type().Nature() == semantic.String {
+					req.Header.Set(k.Str(), v.Str())
+				} else {
+					rangeErr = errors.Newf(codes.Invalid, "header key and values must be a string: %q", k)
+				}
+			})
+			if rangeErr != nil {
+				return nil, rangeErr
+			}
+		}
+
+		// Get Client and configure it
+		deps := flux.GetDependencies(ctx)
+		dc, err := deps.HTTPClient()
+		if err != nil {
+			return nil, errors.Wrap(err, codes.Aborted, "missing client in http.request")
+		}
+
+		timeoutV, ok := config.Get("timeout")
+		if !ok {
+			return nil, errors.New(codes.Invalid, "config is missing \"timeout\" property")
+		}
+		timeout := timeoutV.Duration()
+		if timeout.IsMixed() {
+			return nil, errors.New(codes.Invalid, "config timeout must not be a mixed duration")
+		}
+		dc, err = fhttp.WithTimeout(dc, timeout.Duration())
+		if err != nil {
+			return nil, err
+		}
+
+		insecureSkipVerifyV, ok := config.Get("insecureSkipVerify")
+		if !ok {
+			return nil, errors.New(codes.Invalid, "config is missing \"insecureSkipVerify\" property")
+		}
+		if insecureSkipVerifyV.Bool() {
+			dc, err = fhttp.WithTLSConfig(dc, &tls.Config{
+				InsecureSkipVerify: true,
+			})
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		// Do request, using local anonymous functions to facilitate timing the request
+		statusCode, responseBody, headers, err := func(req *http.Request) (int, []byte, values.Dictionary, error) {
+			s, cctx := opentracing.StartSpanFromContext(req.Context(), "requests._do")
+			s.SetTag("url", req.URL.String())
+			defer s.Finish()
+
+			req = req.WithContext(cctx)
+			response, err := dc.Do(req)
+			if err != nil {
+				// Alias the DNS lookup error so as not to disclose the
+				// DNS server address. This error is private in the net/http
+				// package, so string matching is used.
+				if strings.HasSuffix(err.Error(), "no such host") {
+					return 0, nil, nil, errors.New(codes.Invalid, "no such host")
+				}
+				return 0, nil, nil, err
+			}
+			body, err := ioutil.ReadAll(response.Body)
+			_ = response.Body.Close()
```

Comment:
if `response.Body` is nil, the code will panic with the invalid memory address or nil pointer dereference. So I guess moving this code line after `nil` error check below will make sure that the code returns if there's any error returned from `ioutil.ReadAll` or we can put a nil check on `response.Body` before calling Close on it.

Reply:
@skartikey and I chatted out of band but Go guarantees the Body is non-nil, see https://pkg.go.dev/net/http#Response